optuna/_deprecated.py

```diff
@@ -134,7 +150,7 @@ def wrapped_init(self, *args, **kwargs) -> None:  # type: ignore
 
                 _original_init(self, *args, **kwargs)
 
-            cls.__init__ = wrapped_init
+            setattr(cls, "__init__", wrapped_init)
```

Comment:
https://github.com/python/mypy/issues/2427#issuecomment-384229898
(But some linter is not happy with the hack: https://github.com/python/mypy/issues/2427#issuecomment-1024964171, I'm not sure whether it is preferred or not)